src/Arcus.BackgroundJobs.DataBricks/DatabricksInfoProvider.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Azure.Databricks.Client;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.BackgroundJobs.Databricks
+{
+    /// <summary>
+    /// Provides dev-friendly access to the Databricks instance.
+    /// </summary>
+    public class DatabricksInfoProvider : IDisposable
+    {
+        private readonly DatabricksClient _databricksClient;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of hte <see cref="DatabricksInfoProvider"/> class.
+        /// </summary>
+        /// <param name="databricksClient">The client to interact with Databricks.</param>
+        /// <param name="logger">The instance to log metric reports of job runs.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="databricksClient"/> or <paramref name="logger"/> is <c>null</c>.</exception>
+        public DatabricksInfoProvider(DatabricksClient databricksClient, ILogger logger)
+        {
+            Guard.NotNull(databricksClient, nameof(databricksClient));
+            Guard.NotNull(logger, nameof(logger));
+
+            _databricksClient = databricksClient;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// <para>Measure the Databricks finished jobs by reporting the results as logging metrics.</para>
+        /// <para>
+        ///  Combines:
+        ///  <para><see cref="GetFinishedJobRunsAsync"/> and </para>
+        ///  <para><see cref="ReportFinishedJobOutcomeAsync"/>.</para>
+        /// </para>
+        /// </summary>
+        /// <param name="metricName">The name of the logging metric.</param>
+        /// <param name="startOfWindow">The start of time window which we are interested in. (Inclusive)</param>
+        /// <param name="endOfWindow">The snd of time window which we are interested in. (Exclusive)</param>
+        /// <seealso cref="GetFinishedJobRunsAsync"/>
+        /// <seealso cref="ReportFinishedJobOutcomeAsync"/>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="metricName"/> is blank.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the <paramref name="endOfWindow"/> is less than the <paramref name="startOfWindow"/>.</exception>
+        public async Task MeasureJobOutcomesAsync(string metricName, DateTimeOffset startOfWindow, DateTimeOffset endOfWindow)
```

Comment:
Thought - What if people want to provide contextual information? Should they override or be able to pass an optional dictionary?

Reply:
Moved to logger extension so they can use extension if they want more control on how it is reported.